gpiozero/pins/__init__.py

```diff
@@ -6,39 +8,133 @@
     )
 str = type('')
 
-import io
-
-from .data import pi_info
 from ..exc import (
     PinInvalidFunction,
     PinSetInput,
     PinFixedPull,
+    PinSPIUnsupported,
     PinPWMUnsupported,
     PinEdgeDetectUnsupported,
+    SPIFixedClockMode,
+    SPIFixedBitOrder,
+    SPIFixedSelect,
+    SPIFixedWordSize,
     )
 
 
-PINS_CLEANUP = []
-def _pins_shutdown():
-    for routine in PINS_CLEANUP:
-        routine()
+class Factory(object):
+    """
+    Generates pins, SPI, and I2C interfaces for devices. This is an abstract
+    base class for pin factories. Descendents must override:
+
+    * :meth:`_get_address`
+    * :meth:`pin_address`
```

Comment:
I probably didn't explain myself clearly enough either. I wasn't suggesting changing the doc-strings, but instead swapping around the actual method implementations themselves, so that `def _get_address(self):` and `def pin_address(self, spec):` appear earlier in the file than `def close(self):` (if that makes more sense now?)